src/server/run-websocket-server.ts

```diff
@@ -1,29 +1,54 @@
 import * as WebSocket from "ws";
 import * as Knex from "knex";
+import { EventEmitter } from "events";
 import { JsonRpcRequest } from "../types";
 import { isJsonRpcRequest } from "./is-json-rpc-request";
 import { dispatchJsonRpcRequest } from "./dispatch-json-rpc-request";
 import { makeJsonRpcResponse } from "./make-json-rpc-response";
+import { Subscriptions } from "./subscriptions";
 
 export function runWebsocketServer(db: Knex, port: number): void {
   const websocketServer: WebSocket.Server = new WebSocket.Server({ port });
   websocketServer.on("connection", (websocket: WebSocket): void => {
+    const subscriptions = new Subscriptions();
+
     websocket.on("message", (data: WebSocket.Data): void => {
       try {
         const message = JSON.parse(data as string);
-        if (isJsonRpcRequest(message)) {
+        if (!isJsonRpcRequest(message)) return console.error("bad json rpc message received:", message);
+
+        if (message.method === "subscribe") {
+          const eventName: string = message.params.shift();
```

Comment:
Are there other parameters expected / allowed here, besides the event name?

One important thing to note is that there can be duplicate event names across different contracts; e.g., all ERC-20 contracts have their own `Transfer` and `Approval` events.  Would it be better to key the subscriptions by contract name and event name?  (This is the way augur.js handles event subscriptions internally.)

Reply:
Do we have a use case for needing to specify the contract? I was under the assumption that we were working on these just from the "Augur" perspective, so we're subscribing to MarketCreated as an augur construct, not an arbitrary log.

Reply:
And yes, using the same format as geth for subscriptions https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB.

Ultimately the API supports passing parameters into the subscriptions e.g. to further filter the events delivered.

My internal API supports this, although I don't think we use it for anything.